1ef48c6f3978beedf21b8f70ed101037e68674a7
src/main/java/com/yelp/nrtsearch/server/luceneserver/state/BackendGlobalState.java
--- 
+++ 
@@ -24,6 +24,7 @@
 import com.yelp.nrtsearch.server.grpc.DummyResponse;
 import com.yelp.nrtsearch.server.grpc.GlobalStateInfo;
 import com.yelp.nrtsearch.server.grpc.IndexGlobalState;
+import com.yelp.nrtsearch.server.grpc.IndexLiveSettings;
 import com.yelp.nrtsearch.server.grpc.Mode;
 import com.yelp.nrtsearch.server.grpc.RestoreIndex;
 import com.yelp.nrtsearch.server.grpc.StartIndexRequest;
@@ -129,7 +130,11 @@
     Map<String, IndexStateManager> managerMap = new HashMap<>();
     for (Map.Entry<String, IndexGlobalState> entry : globalStateInfo.getIndicesMap().entrySet()) {
       IndexStateManager stateManager =
-          createIndexStateManager(entry.getKey(), entry.getValue().getId(), stateBackend);
+          createIndexStateManager(
+              entry.getKey(),
+              entry.getValue().getId(),
+              luceneServerConfiguration.getLiveSettingsOverride(entry.getKey()),
+              stateBackend);
       stateManager.load();
       managerMap.put(entry.getKey(), stateManager);
     }
@@ -163,8 +168,11 @@
    * @return index state manager
    */
   protected IndexStateManager createIndexStateManager(
-      String indexName, String indexId, StateBackend stateBackend) {
-    return new BackendStateManager(indexName, indexId, stateBackend, this);
+      String indexName,
+      String indexId,
+      IndexLiveSettings liveSettingsOverrides,
+      StateBackend stateBackend) {
+    return new BackendStateManager(indexName, indexId, liveSettingsOverrides, stateBackend, this);
   }
 
   /**
@@ -188,7 +196,12 @@
       String indexName = entry.getKey();
       IndexStateManager stateManager = immutableState.indexStateManagerMap.get(indexName);
       if (stateManager == null || !entry.getValue().getId().equals(stateManager.getIndexId())) {
-        stateManager = createIndexStateManager(indexName, entry.getValue().getId(), stateBackend);
+        stateManager =
+            createIndexStateManager(
+                indexName,
+                entry.getValue().getId(),
+                getConfiguration().getLiveSettingsOverride(indexName),
+                stateBackend);
       }
       stateManager.load();
       newManagerMap.put(indexName, stateManager);
@@ -262,11 +275,21 @@
     IndexStateManager stateManager;
     if (createIndexRequest.getExistsWithId().isEmpty()) {
       indexId = getIndexId();
-      stateManager = createIndexStateManager(indexName, indexId, stateBackend);
+      stateManager =
+          createIndexStateManager(
+              indexName,
+              indexId,
+              getConfiguration().getLiveSettingsOverride(indexName),
+              stateBackend);
       stateManager.create();
     } else {
       indexId = createIndexRequest.getExistsWithId();
-      stateManager = createIndexStateManager(indexName, indexId, stateBackend);
+      stateManager =
+          createIndexStateManager(
+              indexName,
+              indexId,
+              getConfiguration().getLiveSettingsOverride(indexName),
+              stateBackend);
       stateManager.load();
     }
 